core: invalid cast in opengl.cpp

##### System information (version)
- OpenCV => 3.4
- Operating System / Platform => Windows 64 Bit
- Compiler => mingw64

##### Detailed description
```
C:\p\opencv\modules\core\src\opengl.cpp: In function 'cv::ocl::Context& cv::ogl::ocl::initializeContextFromGL()':
C:\p\opencv\modules\core\src\opengl.cpp:1627:104: error: invalid cast from type 'cv::AutoBuffer<char>' to type 'char*'
                 status = clGetPlatformInfo(platforms[i], CL_PLATFORM_EXTENSIONS, extensionSize, (char*)extensionStr, NULL);
                                                                                                        ^~~~~~~~~~~~
C:\p\opencv\modules\core\src\opengl.cpp:1632:38: error: invalid cast from type 'cv::AutoBuffer<char>' to type 'const char*'
             if (!strstr((const char*)extensionStr, "cl_khr_gl_sharing"))
                                      ^~~~~~~~~~~~
```

probably overlooked, because the buildbots don't have opengl enabled.

--- a/modules/core/src/opengl.cpp
+++ b/modules/core/src/opengl.cpp
@@ -1624,12 +1624,12 @@ Context& initializeContextFromGL()
             if (status == CL_SUCCESS)
             {
                 extensionStr.allocate(extensionSize+1);
-                status = clGetPlatformInfo(platforms[i], CL_PLATFORM_EXTENSIONS, extensionSize, (char*)extensionStr, NULL);
+                status = clGetPlatformInfo(platforms[i], CL_PLATFORM_EXTENSIONS, extensionSize, (char*)extensionStr.data(), NULL);
             }
             if (status != CL_SUCCESS)
                 CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get platform extension string");
 
-            if (!strstr((const char*)extensionStr, "cl_khr_gl_sharing"))
+            if (!strstr((const char*)extensionStr.data(), "cl_khr_gl_sharing"))
                 continue;
         }
 

